CRM/Civicase/Page/ActivityFiles.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+class CRM_Civicase_Page_ActivityFiles {
+
+  /**
+   * Download all activity files contained in a single zip file.
+   */
+  public static function downloadAll() {
+    $activity = self::getActivityFromRequest();
+
+    $zipName = self::getZipName($activity);
+    $zipDestination = self::getDestinationPath();
+    $zipFullPath = $zipDestination . '/' . $zipName;
+    $files = self::getActivityFilePaths($activity['id']);
+    $zipFileResource = self::createZipFile($zipFullPath, $files);
+
+    unlink($zipFullPath);
+    self::downloadZipFileResource($zipName, $zipFileResource);
+  }
+
+  /**
+   * Returns the activity specified by the request. In case the request gives
+   * an invalid activity id it throws a 404 status code.
+   */
+  private static function getActivityFromRequest() {
+    $activityId = CRM_Utils_Array::value('activity_id', $_GET);
+
+    self::validateActivityId($activityId);
+
+    $activityResult = civicrm_api3('Activity', 'get', [
+      'id' => $activityId,
+      'return' => [ 'activity_type_id.label' ]
+    ]);
+
+    if ($activityResult['count'] === 0) {
+      return self::throwStatusCode('404 Not Found');
```

Comment:
Status code is always an integer, isn't it?

Reply:
@igorpavlov it seems CiviCRM is internally using a non standard response header so I decided to revert back to `http_response_code`. @tunbola has also used it for the calendar feeds so I believe it's fine to use it:

https://github.com/compucorp/civihr/blob/d041bad3c40e6d2629b156739d3489efb5150a3f/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Page/LeaveRequestCalendarFeed.php#L25